### PR TITLE
VEN-1000 | Implement winter storage leases and offers

### PR DESCRIPTION
**Pull request by TatuArvela**
### Closes :no_good_woman:

* [VEN-1000](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1000): FE - Assign winter storage place

### Automated tests :gear:️

* Work in progress

### Manual testing :construction_worker_man:

* Handling a winter storage application should work

### CODE REVIEW

**Review by Nizar-Rahme — CHANGES_REQUESTED** [review]
Great job! 
Left a few comments

File: src/features/applicationList/ApplicationList.tsx [hunk 1]
```diff
@@ -185,6 +186,7 @@ const ApplicationList = ({
         renderSubComponent={(row) => (
           <ApplicationDetails
             {...row.original}
+            applicationType={row.original.berthSwitch ? ApplicationTypeEnum.BERTH_SWITCH : ApplicationTypeEnum.BERTH}
```
[the remark this answers is not in the document]
Comment (TatuArvela) [reply src/features/applicationList/ApplicationList.tsx:189]: Yeah, ApplicationList only shows berth applications

File: src/features/winterStoragePlaceDetails/queries.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      width
+      length
+      leases {
+        edges {
+          node {
+            id
+            customer {
+              id
+              firstName
+              lastName
```
Comment (Nizar-Rahme) [inline src/features/winterStoragePlaceDetails/queries.ts:18]: Here, fetching customer's info separately using `CustomerName` might speed up the query this gets slow

File: src/features/customerView/utils.ts [hunk 3]
```diff
@@ -201,6 +202,7 @@ export const getApplications = (profile: CUSTOMER_PROFILE, boatTypes: BOAT_TYPES
         const applicationData = {
           id,
           applicationCode,
+          applicationType: berthSwitch ? ApplicationTypeEnum.BERTH_SWITCH : ApplicationTypeEnum.BERTH,
```
[the remark this answers is not in the document]
Comment (TatuArvela) [reply src/features/customerView/utils.ts:205]: Yep, there's no querying for winter storage applications here. We could add a ticket and add it if it's needed

File: src/features/winterStoragePlaceDetails/WinterStoragePlaceDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    .filter((lease) => !lease.isActive)
+    .map(({ startDate, endDate, customer }, i) => {
+      return (
+        <div key={i}>
+          <Text>{`${formatDate(startDate, i18n.language)} - ${formatDate(endDate, i18n.language)}`}</Text>{' '}
+          <InternalLink to={`/customers/${customer.id}`}>
+            {customer.firstName !== '' && customer.lastName !== ''
+              ? `${customer.firstName} ${customer.lastName}`
+              : t('common.emptyName')}
+          </InternalLink>
```
[the remark this answers is not in the document]
Comment (TatuArvela) [reply src/features/winterStoragePlaceDetails/WinterStoragePlaceDetails.tsx:40]: 🤔 I think this is ok for this component, as it's only used for expanded rows. Only the customers of a single place will be queried at a time

**Review by Nizar-Rahme — APPROVED** [review]
💎

**Pull request merged**